parquet_derive_test/src/lib.rs

```diff
@@ -347,6 +347,36 @@ mod tests {
         assert_eq!(drs[0].isize, out[0].isize);
     }
 
+    #[test]
+    fn test_aliased_result() {
+        // Issue 7547, Where aliasing the `Result` led to
+        // a collision with the macro internals of derive ParquetRecordReader
+
+        mod aliased_result {
+            use parquet_derive::{ParquetRecordReader, ParquetRecordWriter};
+            pub type Result = std::result::Result<(), Box<dyn std::error::Error>>;
+
+            #[derive(ParquetRecordReader, ParquetRecordWriter, Debug)]
+            pub struct ARecord {
+                pub bool: bool,
+                pub string: String,
+            }
+
+            impl ARecord {
+                pub fn validate(&self) -> Result {
```

Comment:
```suggestion
                pub fn validate(&self) -> std::result::Result<(), Box<dyn std::error::Error>> {
```

(so it doesn't use our `Result` shadow)